### PR TITLE
Fix for case when animated joints are missing from the target avatar

### PR DESCRIPTION
**Pull request by hyperlogic**
By copying the animation rotations over to the target avatar in absolute frame, rather than relative,
we can properly "combine" animated rotations that aren't in the target avatar skeleton.

For example: If an avatar is missing the RightShoulder joint but has the RightArm joint.  This PR properly combines the relative rotations from both the Shoulder and the Arm onto the arm joint.  On master, only the Arm relative rotation is copied, resulting in an incorrect arm rotation "zombie arms".

### CODE REVIEW

**Comment by hyperlogic** [comment]
https://highfidelity.manuscript.com/f/cases/21080/MasterChief-avatar-has-arms-out-in-front-if-Shoulders-aren-t-mapped
https://highfidelity.manuscript.com/f/cases/edit/20403/Avatar-leans-when-sitting

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17781/HighFidelity-Beta-PR15068-9a2bd87.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17781/HighFidelity-Beta-Interface-PR15068-9a2bd87.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17781/HighFidelity-Beta-PR15068-9a2bd87.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17781/HighFidelity-Quest-Beta-PR15068-9a2bd87.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17781/HighFidelity-Beta-PR15068-9a2bd87.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17781/HighFidelity-Beta-Interface-PR15068-9a2bd87.exe

File: libraries/animation/src/AnimSkeleton.cpp [hunk 1]
```diff
@@ -149,6 +149,17 @@ void AnimSkeleton::convertAbsolutePosesToRelative(AnimPoseVec& poses) const {
     }
 }
 
+void AnimSkeleton::convertRelativeRotationsToAbsolute(std::vector<glm::quat>& rotations) const {
+    // poses start off relative and leave in absolute frame
```
Comment (amantley) [inline libraries/animation/src/AnimSkeleton.cpp]: rotations not poses in the comment

Comment (hyperlogic) [reply libraries/animation/src/AnimSkeleton.cpp]: I updated this comment

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17792/HighFidelity-Beta-PR15068-82b2050.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17792/HighFidelity-Beta-Interface-PR15068-82b2050.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17792/HighFidelity-Beta-PR15068-82b2050.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17792/HighFidelity-Quest-Beta-PR15068-82b2050.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17792/HighFidelity-Beta-PR15068-82b2050.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17792/HighFidelity-Beta-Interface-PR15068-82b2050.exe

**Pull request merged**